api/src/utils/media/xlsx/transformation/xlsx-transform.ts

```diff
@@ -135,17 +135,17 @@ export class XLSXTransform {
     const worksheetJSONWithKey: RowObject[] = [];
 
     for (let i = 0; i < numberOfRows; i++) {
-      const primaryKey = this._getKeyForRowObject(worksheetJSON[i], templateMetaSchema.primaryKey);
+      const primaryKey = this._getKeyForRowObject(worksheetJSON[i], templateMetaSchema.primaryKey, i);
```

Comment:
I don't think we can have all of these changes that add the index, it breaks other functionality that works with keys and doesn't want to have the row number included.

If we need the row number, we need to just add a `_row` bit to the key in the config.